core/src/main/scala/org/apache/spark/util/collection/PartitionedPairBuffer.scala

```diff
@@ -96,5 +96,5 @@ private[spark] class PartitionedPairBuffer[K, V](initialCapacity: Int = 64)
 }
 
 private object PartitionedPairBuffer {
-  val MAXIMUM_CAPACITY = Int.MaxValue / 2 // 2 ^ 30 - 1
+  val MAXIMUM_CAPACITY = (Int.MaxValue - 8) / 2
```

Comment:
maybe worth adding the `Int` type even though it's already an Int.

Also the comment at the line 28 should be changed to `1073741819` i.e. `- 8/2`